# transposable_element

Collaborated project at [Gilad Lab](http://giladlab.uchicago.edu/) and [Stephens Lab](http://stephenslab.uchicago.edu/). 

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*

- [Overview](#overview)
- [Results for ChIP-seq](#results-for-chip-seq)
- [Results for RNA-seq](#results-for-rna-seq)
- [Collaborate with other using git](#collaborate-with-other-using-git)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

## Overview

The basic idea is firstly to get an overview of how the distribution of the H3K9me3 mark differs between the 2 species and secondly whether there are particular transposable elements that are silenced specifically in one species and try to get at a mechanism for this as transposable element silencing is a rapidly evolving phenomenon. My collaborator can do the transposable element analysis in each species separately but we are likely to need help for the cross-species analysis. We would then ideally want to relate any differences in H3K9me3 occupancy with changes in gene expression.

To check our analysis results:
https://simingz.github.io/transposable_element/analysis

## Results for ChIP-seq 
## Results for RNA-seq 

## Collaborate with other using git

We suggest that each collaborator creates his/her own master copy of the repository - 
that is, forking the repository. Then, all changes to the main copy need to go through
a pull request, which can be reviewed by all collaborators.

How to fork a repo: https://help.github.com/articles/fork-a-repo/

How to sync a fork with the remote repository: https://help.github.com/articles/syncing-a-fork/









