---
title: "Home"
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`


## Overview
The basic idea is firstly to get an overview of how the distribution of the H3K9me3 mark differs between the 2 species and secondly whether there are particular transposable elements that are silenced specifically in one species and try to get at a mechanism for this as transposable element silencing is a rapidly evolving phenomenon. My collaborator can do the transposable element analysis in each species separately but we are likely to need help for the cross-species analysis. We would then ideally want to relate any differences in H3K9me3 occupancy with changes in gene expression.

## H3K9me3 ChIP seq

*  [H3K9me3 ChIP sequencing quality control](tutorial.html) 

*  [H3K9me3 differentially enriched regions in human and chimp](single-cell-pca.html)

## RNA seq

*  [RNA sequencing quality control](tutorial.html) 

*  [Differentailly expressed genes in human and chimp](single-cell-pca.html)

## Regulation of TE and gene expression
